tasks/redhat.yml

```diff
@@ -26,27 +26,25 @@
     mode: "0755"
   when: squid_extra_cache_dir|bool
 
-- name: "SELinux: Allow squid to use SNMP UDP port"
-  community.general.seport:
-    ports: 161
-    proto: "udp"
-    setype: "squid_port_t"
-    state: present
-  when: ansible_selinux is false or
-        ansible_selinux.status == "enabled"
-
-- name: "SELinux: Allow squid to use extra cache dir"
-  community.general.sefcontext:
-    ftype: a
-    setype: squid_cache_t
-    target: "{{ squid_cache_dir2['path'] }}(/.*)?"
-    state: "{{ squid_extra_cache_dir | ternary('present', 'absent') }}"
-  when: ansible_selinux is false or
-        ansible_selinux.status == "enabled"
-
-- name: "Set correct context"
-  ansible.builtin.command: restorecon -R {{ squid_cache_dir2["path"] }}
-  changed_when: true
+# Selinux needs to be enabled to run these tasks. 
+# ansible_selinux.status == "Disabled" and ansible_selinux == false will cause the tasks to fail if selinux is disabled.
+- block:
+  - name: "SELinux: Allow squid to use SNMP UDP port"
+    community.general.seport:
+      ports: 161
+      proto: "udp"
+      setype: "squid_port_t"
+      state: present
+  - name: "SELinux: Allow squid to use extra cache dir"
+    community.general.sefcontext:
+      ftype: a
+      setype: squid_cache_t
+      target: "{{ squid_cache_dir2['path'] }}(/.*)?"
+      state: "{{ squid_extra_cache_dir | ternary('present', 'absent') }}"     
+  - name: "Selinux: Set correct context"
+    ansible.builtin.command: restorecon -R {{ squid_cache_dir2["path"] }}
+    changed_when: true
+  when: ansible_selinux.status == "Enabled"
```

Comment:
I think this should do the trick also for very minimal redhat installations

```
- ansible_selinux != False
- ansible_selinux.status == "Enabled"
```

Reply:
This is now fixed and travis is happy too.